File: Pset01/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long long x;
        printf("Number: ");
        x = get_long_long();
    int arr[100], c = 0;
    for(int i = 0 ; x ; i++)
    {
        arr[i]=x%10;
        c++;
        x/=10;
    }
    if(c>5){
    int tempsum = 0;
    for(int i = 1 ; i<c ; i+=2)
    {
        int q = arr[i]*2;
        while(q)
        {
            tempsum+=q%10;
            q/=10;
        }
    }
     for(int i = 0 ; i<c ; i+=2)
    {

            tempsum+=arr[i];

    }
    bool valid;
    if(tempsum%10 == 0 ) valid = true;
    else valid = false;

    if(!valid) printf("INVALID\n");
    else
    {
        if(arr[c-1]==3) printf("AMEX\n");
        else if (arr[c-1]==4) printf("VISA\n");
        else printf("MASTERCARD\n");
    }
    }
    else
        printf("INVALID\n");
}

File: pset06/credit.py
import cs50

credit = cs50.get_int()
str_credit = str(credit)
cnt = 0
for index in range(len(str_credit)):
    if index%2 == 0:
           cnt+=int(str_credit[len(str_credit)-index-1])
    else:
        temp = str(int(str_credit[(len(str_credit)-index)-1])*2)
        cnt = cnt + int(temp)%10 + int(temp)//10
if cnt%10 == 0 and len(str_credit)==15 and (str_credit[0:2]=="34" or str_credit[0:2]=="37"):
    print("AMEX")
elif cnt%10 == 0 and len(str_credit)==16 and (str_credit[0]=="5" and str_credit[1]<="5" and str_credit[1]>"0"):
    print("MASTERCARD")
elif cnt%10 == 0 and (len(str_credit)==13 or len(str_credit)==16) and str_credit[0]=="4":
    print("VISA")
else :
    print("INVALID")
